### ISSUE TITLE
Make "month" and "day" available as conditions, so missions can check for particular holidays

### ISSUE THREAD

**Issue by natangold** [endless-sky/endless-sky#653]
I was writing up some missions to bring Muslim passengers to Earth for the Hajj (and from afterwards). To do so, I've got four events:
```
event "before Hajj"
	date 25 12 3013
	event "during Hajj" 30

event "during Hajj"
	clear "before Hajj"
	event "after Hajj" 5

event "after Hajj"
	clear "during Hajj"
	event "done with Hajj" 14

event "done with Hajj"
	clear "after Hajj"
	event "before Hajj" 305
```
These events should simulate, more or less, the annual pilgrimage and its arrival on the lunar calendar. During the month leading up to the Hajj, the player might get passengers who want a ride out to Earth, based on whether or not the `"before Hajj"` condition is set. Although these missions could be flagged with a `to fail` once the `"before Hajj"` condition is cleared, I don't see a way to give a player insight into when this deadline would be. Is it possible / can we make it possible to use an event as a deadline as in the following example:
```
mission "Pilgrim to Mecca"
	name "Bring a pilgrim to <planet>"
	job
	repeat
	deadline "during Hajj"
	description "This devout Muslim is performing the rite of Hajj, and needs transportation to Mecca on <planet> by <date>."
	passengers 1
	to offer
		has "before Hajj"
		random < 24
	source
		near Sol 1 
		government "Republic" "Free Worlds" "Syndicate" "Neutral"
	destination Earth
	on complete
		payment
		payment 8000
		dialog "You wish your <passenger> a blessed Hajj, and collect your payment of <payment>."
```

**Comment by endless-sky** [comment]
One mild problem is that the date will slowly drift due to leap years making each year an average of 365.25 days long instead of 365.

But of course, the bigger point is that it'd be nice to have recurring events of some sort. I think the best way to do that will be to have some sort of data type that defines a condition flag that is true or false depending on the current date - the mission could then be offered only when that flag is true and fail if it is false.

**Comment by natangold** [comment]
In the case of the Hajj, the date would drift no matter what (with their lunar year running 10-12 days shorter than the Gregorian), but yes, recurring events. After that first year's "Remembrance Day", it'd be neat to have that show up as an annual [invisible] event, especially if players could see slightly different flavours depending on where they are (with paradise worlds treating it like St. Patty's, frontier worlds like Thanksgiving, military planets like just another Tuesday, and Poisonwood like Yom HaShoah (Jewish/Israeli holiday commemorating the Holocaust).

**Comment by LocalGod79** [comment]
the way the above missions/events are coded, there would also be drift due to the events possibly triggering on days the user does not land - so the next events countdown begins late.

i actually started coding a repeating remembrance day, before it clicked...

events triggering delayed events could be useful.

**Comment by endless-sky** [comment]
In commit 7656422 I've added "day", "month", and "year" condition variables that you can check in a mission. That should give you a much easier way to do this (although not, of course, handling the lunar calendar properly).

**Comment by endless-sky** [comment]
Closing this because it's fixed in 0.9.1, which is released now.
